### ISSUE TITLE
[enhancement] Add support for Google Login

### ISSUE THREAD

**Issue by flancast90** [DarkCat09/python-aternos#19]
As the title implies, an auth flow using Google would be beneficial for google-using clients.

**Comment by DarkCat09** [comment]
Hello.  
It's too hard or impossible because of OAuth security.

**Comment by DarkCat09** [comment]
I'll appreciate if you help me with this feature.

**Comment by flancast90** [comment]
> Hello. It's too hard or impossible because of OAuth security.

Not really. Google App Passwords allow a simple username/password login _without_ disabling the Secure Apps features. I think the hard part will be determining the token Aternos uses for the Google Login. However, if they keep you logged in, you could attempt a normal OAuth flow, and then see if it's persistent across Aternos.

**Comment by DarkCat09** [comment]
> determining the token Aternos uses for the Google Login

Aternos' backend is not open-source.  
It doesn't provide any ajax API for logging in with Google.

**While the feature is not implemented, use `Client.from_session` specifying ATERNOS_SESSION cookie as an argument.**

**Comment by flancast90** [comment]
> Aternos' backend is not open-source. It doesn't provide any ajax API for logging in with Google.

Yes, but the args are passed to Google via URL to open the google signin window. If you could determine the token for Google signin, you could use the standard Google OAuth library without any need for Aternos' backend.

**Comment by TheRedstoneRadiant** [comment]
Google Sign-in Button href: https://aternos.org/auth/google-login 

Redirects to https://accounts.google.com/o/oauth2/auth/oauthchooseaccount?response_type=code&access_type=online&client_id=853067141497-24lhsecgk8lm86qat7nolobpn791o2r6.apps.googleusercontent.com&redirect_uri=https%3A%2F%2Faternos.org%2Fauth%2Fgoogle-login&state=...&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email&approval_prompt=force&flowName=GeneralOAuthFlow

client_id=853067141497-24lhsecgk8lm86qat7nolobpn791o2r6.apps.googleusercontent.com
redirect_uri=https://aternos.org/auth/google-login&
